### DISCUSSION TITLE
Why does Clang Make This Warning?

### DISCUSSION THREAD

**Discussion by SsageParuders** [rui314/mold#932] [Q&A]
I use `mold` for my project with clang.
But clang-15 make a warning like this:
```bash
clang-15: warning: argument unused during compilation: '-fuse-ld=mold' [-Wunused-command-line-argument]
```
And this is the result of `readelf -p comment`:
```bash
String dump of section '.comment':
  [     0]  clang version 15.0.6
  [    15]  mold 1.8.0 (a49a201695edd294ed4d97231c9dc5a994275dd2; compatible with GNU ld)
  [    63]  GCC: (Debian 12.2.0-9) 12.2.0
```
It looks like I had succeed to use `mold`.Should I ignore this warning?

---
Btw, I use clang for my project, but why the section of `.comment` had the info about GCC?

Thinks a lot ~

**Comment by rui314** [comment]
I think you added `-fuse-ld=mold` not only to the last clang invocation to create your executable but to every invocation of clang. When you are creating `.o` files, you don't need to add `-fuse-ld=mold`.

**Reply by SsageParuders** [reply]
You are right. 
With `target_link_options` instead of `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=mold")`
```CMake
target_link_options(${PROJECT_NAME} PRIVATE
       -fuse-ld=mold
)
```
The warning losts, and the section of `.comment` had info about `mold`.
Thinks a lot.

**Comment by ZhongRuoyu** [comment]
> Btw, I use clang for my project, but why the section of .comment had the info about GCC?

That may come from any library / object file that you link against. Like this one from GCC:

```plaintext
$ readelf -p .comment `gcc -print-file-name=crtendS.o`

String dump of section '.comment':
  [     1]  GCC: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0

```

**Reply by SsageParuders** [reply]
Yes, I think maybe I used `libstdc++` from GCC.
